jsonpickle/handlers.py

```diff
@@ -215,8 +215,24 @@ def flatten(self, obj, data):
 
     def restore(self, data):
         restore = self.context.restore
-        factory, args = [restore(i, reset=False) for i in data['__reduce__']]
-        return factory(*args)
+        res = [restore(i, reset=False) for i in data['__reduce__']]
+        factory = res.pop(0)
+        args = res.pop(0)
+        obj = factory(*args)
+
+        if len(res) >= 1:
```

Comment:
Thanks for the suggestions. I thought izip_longest is used to join two sequences with a fill value, if one sequence length exceeds the other one. If one pops the list until its empty, it is more difficult to assign the values properly.